rfc/0002-gotrue-unfork.md

```diff
@@ -0,0 +1,38 @@
+- Feature Name: `gotrue_unfork`
+- Start Date: 2021-10-07
+- RFC PR: [supabase/rfcs#0002](https://github.com/supabase/rfcs/pull/0002)
+- Supabase Issue: [supabase/supabase#0000](https://github.com/supabase/supabase/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Our [Auth server](https://github.com/supabase/gotrue) is currently a fork of [GoTrue by Netlify](https://github.com/netlify/gotrue). This RFC explores the possibility of "unforking".
+
+# Motivation
+[motivation]: #motivation
+
+There are several drawbacks to the current fork:
+
+- *Search*: Due to the way GitHub operates, if you search in our repo it will instead search in the fork. This makes it difficult to find issues/code/PRs.
+- *Divergence*: The core codebase is diverging a lot:
```

Comment:
@kangmingtay could you please list a few of the ways that the fork has diverged? It would help to understand the work involved for bringing Netlify's fork up to date

Reply:
Sure, I'll list down the following ways our fork has diverged:

1. Passwordless-sign-up / sign-in mechanisms (e.g. magiclink & sms otp)
2. Support for HCaptcha
3. Support for sign-in with apple, azure, discord, twitch, twitter
4. Support for phone auth with twilio
5. Gotrue migrations run on start 
6. Filter query behind `/admin/users?filter=<searchterm>` uses postgres dialect instead of mysql
7. Fix email change logic (originally broken upstream)
8. Removal of `DB_NAMESPACE` & `OPERATOR_TOKEN` env vars
9. Admin users are authenticated via JWT instead of being stored in the `users` table with `is_super_admin=true`
10. Addition of schema migrations to support phone auth & email change features
11. Support for setting smtp sender name 

If it's good to go, I can make a commit with a new section in the file to list down all of the differences. Also, I might have missed out on some changes between the time we forked the repo and when i joined. Maybe @awalias will be able to fill that gap?

Reply:
@kangmingtay - let's merge these changes then Ant can add any others later 👍